funsor/terms.py

```diff
@@ -1130,7 +1130,8 @@ def unscaled_sample(self, sampled_vars, sample_inputs):
         return Independent(fn, self.reals_var, self.bint_var)
 
     def eager_subs(self, subs):
-        subs = tuple((self.reals_var_bound, v[self.bint_var])
+        subs = tuple((self.reals_var_bound,
+                      to_funsor(v, self.inputs[k])[self.bint_var])
```

Comment:
This fixes a bug where `isinstance(v, str)`, causing `v[self.bint_var]` to error. The solution is to promote `v` to a `Variable`.